### PR TITLE
Created subcategories. And big refactoring tests

### PR DESCRIPTION
**Pull request by marderer**
- convert category to polymorphic
- convert old categories to user categories, use tasks
1. run `rails db:migrate:up VERSION=20180319083646`
2. run `rails user:convert_categories`

- Add service for creating category
- Refactoring tests.
- Work with views. 

https://drive.google.com/file/d/1SDzwr2dzZwCFZWjfJN_TSaz0i5FS8QVy/view?usp=sharing
download 18

### CODE REVIEW

File: spec/controllers/between_categories_transactions_controller_spec.rb [hunk 1]
```diff
@@ -18,8 +18,10 @@
     context 'when valid' do
       let(:amount) { BigDecimal.new(Faker::Number.decimal(3, 2)) }
       let(:amount_from) { BigDecimal.new(10_000) }
-      let(:category_from) { create(:category, user: user, amount: amount_from) }
-      let(:category_to) { create(:category, user: user) }
+      let(:category_from) do
+        create(:main_category, user: user, amount: amount_from, categorizable: user)
```
Comment (sinpai) [inline spec/controllers/between_categories_transactions_controller_spec.rb]: Isn't it useless to type `categorizable: user` here? 
Isn't `:main_category` already cared about it?

File: spec/controllers/categories_controller_spec.rb [hunk 2]
```diff
@@ -40,15 +40,28 @@
 
   describe 'Post #create' do
     context 'when valid' do
+      subject(:create_category) { post :create, params: params }
+
+      let(:title) { Faker::Internet.user_name(5..15) }
+      let(:params) do
+        {
+          category: {
```
Comment (sinpai) [inline spec/controllers/categories_controller_spec.rb]: Maybe it's better to move `{` to a new line 🤔

File: spec/controllers/category_transactions_controller_spec.rb [hunk 3]
```diff
@@ -19,10 +18,12 @@
   describe 'POST #create' do
     context 'when valid' do
       let(:user) { create(:user) }
+      let(:category) { create(:main_category, categorizable: user) }
+      let(:category_transaction) { create(:category_transaction, category: category) }
 
       it 'creates transaction' do
         expect do
-          create(:transaction, :category_transactions, user: user)
+          create(:category_transactions, user: user, transactinable: category_transaction)
```
Comment (sinpai) [inline spec/controllers/category_transactions_controller_spec.rb]: `transactinable: category_transaction` why do you use it here?

File: spec/controllers/categories_controller_spec.rb (hunk 2 above)
Comment (marderer) [reply spec/controllers/categories_controller_spec.rb]: @sinpai I think there is no difference

**Pull request merged**